### PR TITLE
Tcg app view

### PR DESCRIPTION
**Pull request by Blackshadow8910**
adds the base functionality for the main app panel, where all the actual features of the app will be hosted

### CODE REVIEW

**Review by JimmyyNX — COMMENTED** [review]
LGTM!

File: src/main/java/app/Main.java [hunk 1]
```diff
@@ -23,20 +27,27 @@ public static void main(String[] args) {
...
         GUIManager guiManager = new GUIManager();
-        //Card c = pokemonTCGApiUseCase.apiInteractor.searchCards("mega lopunny").get(0);
+
+        LoginViewModel loginViewModel = new LoginViewModel();
+        AppViewModel appViewModel = new AppViewModel();
+
+        LoginView loginView = LoginView.create(loginViewModel, db);
+        AppView appView = AppView.create(appViewModel);
+        
+        guiManager.addView("app", appView);
```
Comment (yujusti) [inline src/main/java/app/Main.java:44]: Seems good

**Review by JimmyyNX — APPROVED** [review]
Seems very good and detailed and good

File: src/main/java/interface_adapters/ViewModel.java [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+    }
+
+    private final PropertyChangeSupport propertyChangeSupport = new PropertyChangeSupport(this);
+
+    public void firePropertyChanged(String propertyName, Object oldValue, Object newValue) {
+        propertyChangeSupport.firePropertyChange(propertyName, oldValue, newValue);
+
+    }
+
+    public void addPropertyChangeListener(PropertyChangeListener listener) {
```
Comment (yujusti) [inline src/main/java/interface_adapters/ViewModel.java:22]: very nice implementation, here

File: src/main/java/view/AppView.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            Dimension d = new Dimension(40, 40);
+            Image i = ImageIO.read(new File("resources/img/user-icon.png")).getScaledInstance(d.width, d.height, Image.SCALE_SMOOTH);
+            userIconLabel.setIcon(new ImageIcon(i));
+            userIconLabel.setPreferredSize(d);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        setLayout(borderLayout);
+        buttonPanel.setPreferredSize(new Dimension(0, 80));
```
Comment (yujusti) [inline src/main/java/view/AppView.java:94]: Did not we could do this!

**Review by yujusti — COMMENTED** [review]
Very good implementation of the app view

**Pull request merged**